upgrade/boot_loader/VerifierEcDsa.hpp

```diff
@@ -10,12 +12,44 @@ namespace application
     {
     public:
         explicit VerifierEcDsa(infra::ConstByteRange key);
+        virtual ~VerifierEcDsa() = default;
 
         bool IsValid(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature, const hal::SynchronousFlash::Range& data) const override;
 
     private:
+        virtual infra::ConstByteRange GetSignatureStorage(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature) const = 0;
+        virtual uECC_Curve GetCurve() const = 0;
```

Comment:
These functions are important for derived classes, so they should be protected.

```suggestion
    protected:
        virtual infra::ConstByteRange GetSignatureStorage(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature) const = 0;
        virtual uECC_Curve GetCurve() const = 0;
```